cpp/SnowflakeS3Client.cpp

```diff
@@ -136,13 +136,7 @@ RemoteStorageRequestOutcome SnowflakeS3Client::upload(FileMetadata *fileMetadata
               s3Client->HeadObject(headObjectRequest);
 
       if (outcome.IsSuccess()) {
-          std::string sfcDigest = outcome.GetResult().GetMetadata().at(
```

Comment:
This also fixes the crash when metadata does not have SFC_DIGEST. 
"Further more, I found in following case the server doesn’t return “sfc-digest” in the result which crashes the driver.

 

create temporary stage sen.harrystagetest

copy into @sen.harrystagetest from sen.varchar_table file_format=(FIELD_OPTIONALLY_ENCLOSED_BY='"')

get @sen.harrystagetest file://c:\snowflake\get

put file://c:\snowflake\get\data_0_0_0.csv.gz @sen.harrystagetest OVERWRITE = false

 

The checksum should be the same because the upload file is downloaded by get command. If the existing file is generated by copy into command, the server doesn’t return “sfc-digest” and the driver doesn’t check if “sfc-digest” exists before get the value. That’s why the driver crashes in such case.

Is it a server issure not returning “sfc-digest”? I’ll fix the driver crash by checking existence of “sfc-digest” before get the value."